src/AWS.Deploy.Common/Recipes/OptionSettingItem.ValueOverride.cs

```diff
@@ -102,13 +102,16 @@ public async Task SetValue(IOptionSettingHandler optionSettingHandler, object va
                     var result = await validator.Validate(valueOverride);
                     if (!result.IsValid)
                     {
-                        throw new ValidationFailedException(DeployToolErrorCode.OptionSettingItemValueValidationFailed,
-                            result.ValidationFailedMessage?.Trim() ?? $"The value '{valueOverride}' is invalid for option setting '{Name}'.");
+                        Validation.ValidationStatus = ValidationStatus.Invalid;
+                        Validation.ValidationMessage = result.ValidationFailedMessage?.Trim() ?? $"The value '{valueOverride}' is invalid for option setting '{Name}'.";
+                        Validation.InvalidValue = valueOverride;
+                        throw new ValidationFailedException(DeployToolErrorCode.OptionSettingItemValueValidationFailed, Validation.ValidationMessage);
                     }
                 }
                 
                 Validation.ValidationStatus = ValidationStatus.Valid;
                 Validation.ValidationMessage = string.Empty;
+                Validation.InvalidValue = null;
             }
 
             if (AllowedValues != null && AllowedValues.Count > 0 && valueOverride != null &&
```

Comment:
Should an invalid value due to the `AllowedValues` list also set the new property?

And it's not this code path, but similarly wondering about recipe validators...

Reply:
Fixed